client/pages/orders/[orderId].js

```diff
@@ -289,12 +291,16 @@ export async function getServerSideProps(context) {
       },
     };
   } else {
-    const { data: allMyOrder } = await client.get(`/api/orders/myorders`);
+    const { data: allMyOrder } = await client
+      .get(`/api/orders/myorders`)
+      .catch((err) => {
+        console.log(err.message);
+      });
 
     const existOrder = allMyOrder.find((order) => order.id === orderId);
 
     // If userId not match with userId in the order AND user is not an admin
-    if (existOrder === undefined && data.currentUser.isAdmin === false) {
+    if (!existOrder && data.currentUser.isAdmin === false) {
```

Comment:
`find()` method on an array will return `undefined` if the condition is not satisfied.

Might need to re-check if-condition in case existOrder is `undefined` and the user is not an admin.